pkg/controller/storageprofile-controller.go

```diff
@@ -44,8 +54,24 @@ func (r *StorageProfileReconciler) Reconcile(_ context.Context, req reconcile.Re
 	if err := r.client.Get(context.TODO(), req.NamespacedName, storageClass); err != nil {
 		return reconcile.Result{}, err
 	}
+	if _, err := r.reconcileStorageProfile(storageClass); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Alert for incomplete profiles
```

Comment:
I think this code should be in a different function and not in the reconcile function like: updateIncompleteProfiles

Reply:
or checkIncompleteProfiles

Reply:
Done